aptible/resource_endpoint.go

```diff
@@ -113,6 +124,27 @@ func resourceEndpoint() *schema.Resource {
 	}
 }
 
+func resourceEndpointValidate(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+	d := ResourceDiff{ResourceDiff: diff}
+	interfaceContainerPortsSlice := d.Get("container_ports").([]interface{})
+	containerPorts, _ := aptible.MakeInt64Slice(interfaceContainerPortsSlice)
+	containerPort, _ := (d.Get("container_port").(int))
+	endpointType := d.Get("endpoint_type").(string)
+	var err error
+
+	// container_port and container ports are mutually exclusive fields
+	if containerPort != 0 && len(containerPorts) != 0 {
+		err = multierror.Append(err, fmt.Errorf("do not specify container ports AND container port (see terraform docs)"))
+	}
```

Comment:
In the unlikely event that these values aren't known before apply, they'll default to `0` and `[]` which will pass this validation. Then, I believe, the apply will fail when the API returns a validation error. I don't have a problem with this process I just wanted to make sure we were clear on what happens in this case and are OK with it.

Reply:
ex below (i alternated the container_port // container_ports) without fail

<img width="391" alt="image" src="https://github.com/aptible/terraform-provider-aptible/assets/2961973/5713f65e-90ec-48f1-a446-05970a1fbd38">

Reply:
Sorry, I didn't describe the scenario fully and I'm now realizing that it's a non-issue here. I'm not talking about changing values but rather situations where the value won't be known until after apply:

> NOTE: CustomizeDiff does not currently support computed/"known after apply" values from other resource attributes.
\- https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/customizing-differences

In the event that the user provides both a `contianerPort` and `containerPorts` but one or both of them aren't known until after apply, the provider won't catch the error.

This situation caused problems for me previously but I was trying to implement conditionally required fields. In that case, using a computed/"known after apply" value caused the diff to error even if a value would have been provided on apply. In this case, the provider won't catch the error but on apply the API will return a validation error which is fine especially since the alternative is to put the provider validation in the apply.

Reply:
the additional guard we have from switching back and for this now is the https / tcp + tls mutual exclusion, it forces new and is a better experience it seems